src/pages/library/common.js

```diff
@@ -48,6 +48,44 @@ const groupByFeaturedTags = (workspaces, sidebarSections) => _.flow(
   _.fromPairs
 )(sidebarSections)
 
+const numLabelsToRender = 5
+// Takes the top n labels and appends any labels selected by the user to the list
+// When filter options are hidden (e.g. long lists), this will keep user selected items in view
+const computeLabels = (allLabels, selectedLabels) => _.flow(
+  _.intersection(allLabels),
+  _.concat(_.take(numLabelsToRender, allLabels)),
+  _.uniq
+)(selectedLabels)
+
+const FilterSection = ({ onTagFilter, labels, selectedTags, labelRenderer, listDataByTag }) => {
+  // State
+  const [showAll, setShowAll] = useState(false)
+  const lowerSelectedTags = _.map('lowerTag', selectedTags)
+  const labelsToDisplay = showAll ? labels : computeLabels(labels, _.map('label', selectedTags))
+
+  //Render
```

Comment:
I judge that these section headings are overkill.

Reply:
@zarsky-broad Has been adding these headings as markers for folks who are newer to the codebase / don't work in it too often as indicators of the structure of the code. I am going to leave these in as I am following this pattern.

Reply:
I find those comments helpful as well, especially in long files.